crates/democracy/src/lib.rs

```diff
@@ -1091,6 +1110,34 @@ impl<T: Config> Pallet<T> {
         Ok(weight)
     }
 
+    /// determine whether or not a new referendum should be launched. This will return true
+    /// once every week.
+    fn should_launch(now: Duration) -> Result<bool, Error<T>> {
+        if now.as_secs() < NextLaunchTimestamp::<T>::get() {
+            return Ok(false);
+        }
+
+        // time to launch - calculate the date of next launch.
+
+        // convert to format used by `chrono`
+        let secs: i64 = now.as_secs().try_into()?;
+        let now =
+            chrono::NaiveDateTime::from_timestamp_opt(secs, now.subsec_nanos()).ok_or(Error::<T>::TryIntoIntError)?;
+
+        // calculate next week boundary
+        let beginning_of_week = now.date().week(chrono::Weekday::Mon).first_day();
```

Comment:
IMO the date and time should be parameterizable.

Reply:
> I would prefer to keep LaunchPeriod and use that to store chrono::Weekday::Mon.

That doesn't make sense - it wouldn't be a launch period, the period would always be 7 days. It would only change the launch weekday.

> IMO the date and time should be parameterizable.

I disagree - why complicate things unnecessarily? Imo it's better to keep things as simple as possible

Reply:
My primary dislike is that it is effectively a magic number buried in the code, can we at least make it a constant?

Reply:
It would be good if we could read the time somehow via polkadot.js e.g., as a constant because otherwise subsquare or others might need to hardcode it cand can't get it programmatically easily.

Having it configurable would also be OK for me but not a strict requirement.